simphony/io/tests/test_h5_cuds.py

```diff
@@ -255,5 +259,85 @@ def tearDown(self):
             engine.close()
 
 
+class TestMaterialsOperations(unittest.TestCase):
+
+    def setUp(self):
+        self.temp_dir = tempfile.mkdtemp()
+        self.filename = os.path.join(self.temp_dir, 'test.cuds')
+        self.addTypeEqualityFunc(
+            Material, partial(compare_materials, testcase=self))
+
+        self.expected = Materials()
+        for i in xrange(5):
+            self.expected.add_material(
+                Material(description="Material {}".format(i),
+                         data=create_data_container()))
+
+    def tearDown(self):
+        shutil.rmtree(self.temp_dir)
+
+    def file_factory(self):
+        return H5CUDS.open(self.filename)
+
+    def cleanup(self):
+        os.remove(self.filename)
+
+    def test_add_get_materials(self):
+        handle = self.file_factory()
+        expected_materials_dict = {}
+        for material in self.expected.iter_materials():
+            expected_materials_dict[material.uid] = material
+
+        handle.add_materials(self.expected)
+
+        # Closing and reopening the file
+        handle.close()
+        handle = self.file_factory()
+
+        stored = handle.get_materials()
+
+        stored_materials = [material for material in stored.iter_materials()]
+
+        self.assertEqual(len(expected_materials_dict), len(stored_materials))
+        for material in stored_materials:
+            self.assertEqual(
+                material, expected_materials_dict[material.uid])
+
+        handle.close()
+        self.cleanup()
+
+    def test_remove_materials(self):
+        handle = self.file_factory()
+        handle.add_materials(self.expected)
+
+        handle.remove_materials()
+
+        # Closing and reopening the file
+        handle.close()
+        handle = self.file_factory()
+
+        stored = handle.get_materials()
+        stored_materials = [material for material in stored.iter_materials()]
+
+        # The file should not contain any materials
+        self.assertEqual(len(stored_materials), 0)
+
+        handle.close()
+        self.cleanup()
+
+    def test_error_when_adding_existing_materials(self):
+        handle = self.file_factory()
+        handle.add_materials(self.expected)
+
+        handle.close()
+        handle = self.file_factory()
+
+        with self.assertRaises(ValueError):
+            handle.add_materials(self.expected)
+
+        handle.close()
+        self.cleanup()
+
+
```

Comment:
At the moment the distinction between `test_h5_cuds` and `test_h5_*` is not very clear to me. Shouldn't this test goes into `test_h5_materials`? If it is in the right place, we might keep a convention of naming classes in this file as `Test----CudsOperations'.